osdep/terminal-unix.c

```diff
@@ -390,6 +393,7 @@ static void close_tty(void)
 
 static void quit_request_sighandler(int signum)
 {
+    terminal_show_cursor(true);
     do_deactivate_getch2();
 
     (void)write(death_pipe[1], &(char){1}, 1);
```

Comment:
To avoid calling `printf` in the signal handler, could use whatever event is triggered by writing to the `death_pipe` instead.